base/gfspclient/signer.go

```diff
@@ -588,3 +588,91 @@ func (s *GfSpClient) DeleteGlobalVirtualGroup(ctx context.Context, deleteGVG *vi
 	}
 	return resp.GetTxHash(), nil
 }
+
+func (s *GfSpClient) ReserveSwapIn(ctx context.Context, reserveSwapIn *virtualgrouptypes.MsgReserveSwapIn) (string, error) {
+	conn, err := s.SignerConn(ctx)
+	if err != nil {
+		log.Errorw("failed to connect to signer", "error", err)
+		return "", ErrRPCUnknownWithDetail("client failed to connect to signer, error: ", err)
+	}
+	req := &gfspserver.GfSpSignRequest{
+		Request: &gfspserver.GfSpSignRequest_ReserveSwapIn{
+			ReserveSwapIn: reserveSwapIn,
+		},
+	}
+	resp, err := gfspserver.NewGfSpSignServiceClient(conn).GfSpSign(ctx, req)
+	if err != nil {
+		log.CtxErrorw(ctx, "client failed to sign reject migrate bucket", "msg", reserveSwapIn, "error", err)
+		return "", ErrRPCUnknownWithDetail("client failed to sign reject migrate bucket, error: ", err)
+	}
+	if resp.GetErr() != nil {
+		return "", resp.GetErr()
+	}
+	return resp.GetTxHash(), nil
+}
+
+func (s *GfSpClient) CompleteSwapIn(ctx context.Context, completeSwapIn *virtualgrouptypes.MsgCompleteSwapIn) (string, error) {
+	conn, err := s.SignerConn(ctx)
+	if err != nil {
+		log.Errorw("failed to connect to signer", "error", err)
+		return "", ErrRPCUnknownWithDetail("client failed to connect to signer, error: ", err)
+	}
+	req := &gfspserver.GfSpSignRequest{
+		Request: &gfspserver.GfSpSignRequest_CompleteSwapIn{
+			CompleteSwapIn: completeSwapIn,
+		},
+	}
+	resp, err := gfspserver.NewGfSpSignServiceClient(conn).GfSpSign(ctx, req)
+	if err != nil {
+		log.CtxErrorw(ctx, "client failed to sign reject migrate bucket", "msg", completeSwapIn, "error", err)
+		return "", ErrRPCUnknownWithDetail("client failed to sign reject migrate bucket, error: ", err)
+	}
+	if resp.GetErr() != nil {
+		return "", resp.GetErr()
+	}
+	return resp.GetTxHash(), nil
+}
+
+func (s *GfSpClient) SpExit(ctx context.Context, spExit *virtualgrouptypes.MsgStorageProviderExit) (string, error) {
```

Comment:
As "sp exit" is a very general term which can cover a large range of topics, can we have some comments for this methods?

So does this SpExit method mean to declare a certain SP wants to gracefully exit?